app/controllers/teams_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TeamsController < ApplicationController
+  before_action :set_team, only: [:show, :edit, :update, :destroy, :join, :leave]
+
+  def index
+    @teams = Team.all
+  end
+
+  def show
+  end
+
+  def new
+    @team = Team.new
+  end
+
+  def create
+    @team = Team.new(team_params)
+
+    if @team.save
+      redirect_to @team, notice: 'Team was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @team.update(team_params)
+      redirect_to @team, notice: 'Team was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+  def join
+    if @team.users << current_user
+      redirect_to @team, notice: 'Team was successfully joined.'
+    else
+      render :show
```

Comment:
Should we return an error message in this case?

Reply:
This is the default Rails CRUD way for an app with Rails views. We can probably implement a way to let the user know why the user couldn't join, but since we don't have such mechanism currently, I'd leave that as a follow-up.